meetings/2022-11/dec-01.md

```diff
@@ -0,0 +1,1104 @@
+
+JWK: Yes. Dynamic dispatch will be slower. But it looks like the committee has not been convinced by the motivation, so we can discuss this later if anyone is interested.
+
+BT: I think SFC has a good question that that might indicate some next steps here. SFC, if he wanted. to say more about that.
+
+SFC: Yes. So I think that the research call that YSV runs every month, this seems like a good venue to explore this because really this is about how JavaScript developers see and interpret promises and we know we want to make sure that promises are done correctly, and that we avoid foot guns. I think I've heard people in this committee previously discuss how async/await already created some foot guns for developers. It's not clear whether this proposal would help or hurt. I think it would make the proposal stronger if there's some evidence from developers, and a really good way to get that is to, you know, engage in user research.
+
+JWK: That's a good suggestion. Thank you. I will try to bring this topic up in the research call.
+
+BT: Sorry newt. there's one more item on the Queue, RPY . do you want to go ahead?
+
+RPY: Kind of, a continuation of what SFC was saying, and so my own feeling. So if the goal here is to hide the complexity of promises then I think this is to surface level because it's really just renaming utilities that are there. I think maybe explore where users are finding that complexity and what more we could do to make it easier for people to not have to be exposed to promises. Because right now, like as soon as you debug these objects or anything you're going to be exposed to promises.
+
+JWK: I think there isn't much space we can further hide the promises, because for things already async you don't need a promise but if things are not, for example from event emitter it has to call the promise constructor.
+
+RPY: Okay, in that case, you may as well deliberately force users to learn promises and understand how to use them,
+
+JWK: Hmm. Okay. Thanks.
+
+JHD: I have a comments about that. So for me, this proposal is not about hiding promises in any way.  I'm technically a co-champion this proposal and I think that that motivation is actually a reason not to have it. For me it's about the ergonomics of using promises, and reaching await and being able to use await.all For example, instead of having to kind of break the chain, when you do promise.all you have to do the thing with the parentheses that admittedly with something like pipeline, you would it would be a more ergonomic. My pipeline doesn't exist, yet to me this is like the same. A lot of the foot guns that I see in a way async await usage are caused by the attractiveness of using await. Unnecessarily. And when these options syntactic options are available than they become equally attractive and often they are the more correct Choice than await.
```

Comment:
```suggestion
JHD: I have a comments about that. So for me, this proposal is not about hiding promises in any way.  I'm technically a co-champion this proposal and I think that that motivation is actually a reason not to have it. For me it's about the ergonomics of using promises, and reaching for `await` and being able to use `await.all`, for example, instead of having to kind of break the chain. When you do `Promise.all` you have to do the thing with the parentheses  -that admittedly with something like pipeline, would be a more ergonomic, but pipeline doesn't exist yet. A lot of the footguns that I see in `async`/`await` usage are caused by the attractiveness of using `await` unnecessarily. And when these syntactic options are available than they become equally attractive and often they are the more correct choice than `await`.
```